mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```
@@ -0,0 +1,425 @@
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanDeaths() + toAdd;
+
+        rec.setClanDeaths(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_DEATHS);
+
+        this.clanDeaths = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Adds assists to a gg.solarmc.loader.clans.Clan.
+     * @param transaction tx
+     * @param toAdd the amount you want to add
+     * @return Accurate assists after transaction
+     */
+    public Integer addAssists(Transaction transaction, int toAdd) {
+        ClansClanInfoRecord rec = getInformation(transaction);
+
+        int newValue = rec.getClanAssists() + toAdd;
+
+        rec.setClanAssists(newValue);
+        rec.store(CLANS_CLAN_INFO.CLAN_ASSISTS);
+
+        this.clanAssists = newValue;
+
+        return newValue;
+    }
+
+    /**
+     * Gets all clan members accurately
+     * @param transaction the tx
+     * @return the clan members
+     */
+    public Set<ClanMember> getClanMembers(Transaction transaction) {
+        return transaction.getProperty(DSLContext.class)
+                .select(CLANS_CLAN_MEMBERSHIP.USER_ID)
+                .from(CLANS_CLAN_MEMBERSHIP)
+                .where(CLANS_CLAN_MEMBERSHIP.CLAN_ID.eq(this.clanID))
+                .fetchSet((rec) -> new ClanMember(this.clanID,rec.value1(),manager));
+    }
+
+    /**
+     * Adds a member to the clan. Note that this does not have a built in limit, API implementer needs to implement
+     * limit if we desire a limit on members
+     *
+     * Note to API implementer: This handles nothing regarding invites. If you want an invite system, it needs to come
+     * before this method.
+     *
+     * @param transaction The tx
+     * @param player player to add
+     * @return Accurate set of ClanMembers post transaction
+     * @throws IllegalStateException if member is already a member of another clan
+     * @throws IllegalArgumentException if member is the leader of the clan
```

We should throw the same exception to be consistent. IllegalArgumentException seems most suitable to the context.